fastlane/lib/fastlane/actions/set_build_number_repository.rb

```diff
@@ -39,17 +42,24 @@ def self.available_options
                                        description: "Use hg revision number instead of hash (ignored for non-hg repos)",
                                        optional: true,
                                        is_string: false,
-                                       default_value: false)
+                                       default_value: false),
+          FastlaneCore::ConfigItem.new(key: :xcodeproj,
+                                       env_name: "XCODEPROJ_PATH",
+                                       description: "explicitly specify which xcodeproj to use",
+                                       optional: true)
```

Comment:
The following would be good to add to the `ConfigItem` for some value validation:

```ruby
verify_block: proc do |value|
  path = File.expand_path(value)
  UI.user_error!("Please pass the path to the project, not the workspace") if path.end_with?(".xcworkspace")
  UI.user_error!("Could not find Xcode project at #{path}") unless Helper.is_test? || File.exist?(path)
end
```